components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/security/APIAuthenticationHandler.java

```diff
@@ -172,52 +193,88 @@ protected APIManagerConfigurationService getApiManagerConfigurationService() {
     }
 
     public void destroy() {
-        if (authenticator != null) {
-            authenticator.destroy();
+        if (authenticators != null) {
```

Comment:
Eliminate null check by initialization of list during object construction as stated earlier

Reply:
Updated